packages/core/src/storages/key_value_store.ts

```diff
@@ -213,6 +230,23 @@ export class KeyValueStore {
         // Make copy of options, don't update what user passed.
         const optionsCopy = { ...options };
 
+        // If we try to set the value of a cached state to a different reference, we need to update the cache accordingly.
+        const cachedValue = this.cache.get(key) as T;
+
+        if (key in this.cache && cachedValue !== value) {
```

Comment:
Did you mean this.cache.has(key)?

Reply:
good catch, i had a POJO there initially